node_modules/col-core/lib/db.js

```diff
@@ -78,19 +78,323 @@ var init = module.exports.init = function(callback) {
+      'unique': true
+    },
+    'use_https': {
+      'type': Sequelize.BOOLEAN,
+      'defaultValue': true
+    }
+  }, {
+    'underscored': true
+  });
+
+  /**
+   * The `course` table keeps track of each course where the tools have been embedded in
+   *
+   * @property  {Number}      canvas_course_id        The id of the course in Canvas
+   * @property  {Boolean}     enable_upload           Whether or not students are allowed to upload assets directly
+   * @property  {Boolean}     enable_sync             Whether or not this course's data should be synced with Canvas
+   */
+  var Course = module.exports.Course = sequelize.define('course', {
+    'canvas_course_id': {
+      'type': Sequelize.INTEGER,
+      'allowNull': false
+    },
+    'enable_upload': {
+      'type': Sequelize.BOOLEAN,
+      'allowNull': false,
+      'defaultValue': true
+    },
+    'enable_sync': {
+      'type': Sequelize.BOOLEAN,
+      'allowNull': false,
+      'defaultValue': true
+    }
+  }, {
+    'underscored': true
+  });
+
+  // Each course belongs to a specific Canvas instance
+  Course.belongsTo(Canvas, {'onDelete': 'CASCADE'});
+
+  /**
+   * The `user` table keeps track of each user that is enrolled in a course. This means that
+   * the same physical student might end up with two student records within Collabosphere. As
+   * there's no overlap between courses, this is acceptable.
+   *
+   * @property  {Number}        canvas_user_id          The id of the user in Canvas
+   * @property  {String}        canvas_course_role      The role of the user in the course
+   * @property  {String}        canvas_full_name        The full name of the student
+   * @property  {String}        canvas_image            The URL that points to an image for the user
+   * @property  {Number}        points                  The total points that the user has accumulated in the course
+   * @property  {Boolean}       share_points            Whether the user wants to share their point total with the other users in the course
+   */
+  var User = module.exports.User = sequelize.define('user', {
+    'canvas_user_id': {
+      'type': Sequelize.INTEGER,
+      'allowNull': false
+    },
+    'canvas_course_role': {
+      'type': Sequelize.STRING,
+      'allowNull': false
+    },
+    'canvas_full_name': {
+      'type': Sequelize.STRING,
+      'allowNull': false
+    },
+    'canvas_image': {
+      'type': Sequelize.STRING,
+      'allowNull': true
+    },
+    'points': {
+      'type': Sequelize.INTEGER,
+      'allowNull': false,
+      'defaultValue': 0
+    },
+    'share_points': {
+      'type': Sequelize.BOOLEAN,
+      'allowNull': true,
+      'defaultValue': null
+    }
+  }, {
+    'underscored': true
+  });
+
+  // Each user belongs to a specific course
+  User.belongsTo(Course, {'onDelete': 'CASCADE'});
+
+  // TODO: Define these and abstract them out in constants
+  var activityTypes = ['like', 'dislike', 'comment', 'submit_assignment'];
```

Comment:
Hence the `TODO`. I'll come back to this once the module structure has been set up. We can look into abstracting these out into proper constants